### PR TITLE
Added B-logic to sign-in component

### PR DESCRIPTION
**Pull request by KimAT25**
Finished with adding B-logic to sign-in component. 
Added correction to language translation json files.

### CODE REVIEW

File: src/app/component/auth/sign-in-new/sign-in-new.component.html [hunk 1]
```diff
@@ -32,12 +68,30 @@ <h2>{{'user.auth.sign-in.fill-form' | translate}}</h2>
...
+        </div>
+      </div>
+      <div class="validation-password-error"
+           *ngIf="passwordErrorMessageBackEnd!=null">
+        {{passwordErrorMessageBackEnd}}
+      </div>
+      <div class="alert-general-error"
+           *ngIf="backEndError!=null">
+        {{backEndError}}
+      </div>
```
Comment (nabermi2020) [inline src/app/component/auth/sign-in-new/sign-in-new.component.html:87]: On line 119 please add appropriate functionality to open 'Sign up' form

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts [hunk 2]
```diff
@@ -19,20 +21,91 @@ export class SignInNewComponent implements OnInit {
...
     private router: Router,
     private authService: AuthService,
-    private googleService: GoogleSignInService
+    private googleService: GoogleSignInService,
+    private localStorageService: LocalStorageService,
   ) { }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.userOwnSignIn = new UserOwnSignIn();
```
Comment (nabermi2020) [inline src/app/component/auth/sign-in-new/sign-in-new.component.ts:40]: Please divide this code into several methods and add subscription

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts [hunk 3]
```diff
@@ -19,20 +21,91 @@ export class SignInNewComponent implements OnInit {
...
+    this.passwordErrorMessageBackEnd = null;
+    this.backEndError = null;
+    this.localStorageService.userIdBehaviourSubject.subscribe(userId => {
+      if (userId) {
+        this.matDialogRef.close();
+      }
+    });
+  }
+
+  public signIn(userOwnSignIn: UserOwnSignIn) {
```
Comment (nabermi2020) [inline src/app/component/auth/sign-in-new/sign-in-new.component.ts:52]: Please specify return type

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts [hunk 4]
```diff
@@ -19,20 +21,91 @@ export class SignInNewComponent implements OnInit {
...
+    this.loadingAnim = true;
+    this.userOwnSignInService.signIn(userOwnSignIn).subscribe(
```
Comment (nabermi2020) [inline src/app/component/auth/sign-in-new/sign-in-new.component.ts:54]: Please add 2 callbacks

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts [hunk 5]
```diff
@@ -19,20 +21,91 @@ export class SignInNewComponent implements OnInit {
...
+      }
+    );
+  }
 
-  private closeSignInWindow(): void {
+  private CloseSignInWindow(): void {
     this.matDialogRef.close();
   }
 
+  signInWithGoogle() {
```
Comment (nabermi2020) [inline src/app/component/auth/sign-in-new/sign-in-new.component.ts:85]: Please add access modifiers and specify return type

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts [hunk 6]
```diff
@@ -19,20 +21,91 @@ export class SignInNewComponent implements OnInit {
...
+    this.authService.signIn(GoogleLoginProvider.PROVIDER_ID).then(data => {
+      this.googleService.signIn(data.idToken).subscribe(
```
Comment (nabermi2020) [inline src/app/component/auth/sign-in-new/sign-in-new.component.ts:87]: please add 2 callbacks

File: src/app/component/auth/sign-in-new/sign-in-new.component.html (hunk 1 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.html:87]: Fixed

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts (hunk 3 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.ts:52]: Fixed

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts (hunk 5 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.ts:85]: Fixed

File: src/app/component/auth/sign-in-new/sign-in-new.component.html [hunk 7]
```diff
@@ -5,25 +5,61 @@
          alt="sign-in image">
   </div>
   <div class="right-side">
-    <a (click)="closeSignInWindow()"
+    <a (click)="CloseSignInWindow()"
```
Comment (nvolodymyr) [inline src/app/component/auth/sign-in-new/sign-in-new.component.html:8]: add "on" for all click events. It more readable-> "onCloseSignInWindow()"

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts [hunk 8]
```diff
@@ -1,40 +1,141 @@
...
+import { Component, OnInit, OnDestroy } from '@angular/core';
 import { AuthService, GoogleLoginProvider } from 'angularx-social-login';
 import { GoogleSignInService } from '../../../service/auth/google-sign-in.service';
 import { UserSuccessSignIn } from '../../../model/user-success-sign-in';
 import { HttpErrorResponse } from '@angular/common/http';
 import { UserOwnSignInService } from '../../../service/auth/user-own-sign-in.service';
 import { Router } from '@angular/router';
 import { SignInIcons } from 'src/assets/img/icon/sign-in/sign-in-icons';
-import { MatDialogRef } from '@angular/material';
+import {MatDialog, MatDialogRef} from '@angular/material';
```
Comment (nvolodymyr) [inline src/app/component/auth/sign-in-new/sign-in-new.component.ts:9]: add spaces

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts (hunk 2 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.ts:43]: Fixed

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts (hunk 4 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.ts:66]: Fixed

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts (hunk 6 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.ts:78]: Fixed

File: src/app/component/auth/sign-in-new/sign-in-new.component.html (hunk 7 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.html:8]: Actually it`s not a requirement to add "on"

File: src/app/component/auth/sign-in-new/sign-in-new.component.ts (hunk 8 above)
Comment (KimAT25) [reply src/app/component/auth/sign-in-new/sign-in-new.component.ts]: Fixed

**Pull request merged**